test/FunctionalTests/Microsoft.OData.Core.Tests/Json/DefaultJsonWriterFromStreamFactoryTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DefaultJsonWriterFromStreamFactoryTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETCOREAPP3_1_OR_GREATER
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.OData.Json;
+using Xunit;
+
+namespace Microsoft.OData.Tests.Json
+{
+    public class DefaultJsonWriterFromStreamFactoryTests
+    {
+        [Fact]
+        public void CreatesJsonWriterWithUtf8Support()
+        {
+            DefaultStreamBasedJsonWriterFactory factory = DefaultStreamBasedJsonWriterFactory.Instance;
+            MemoryStream stream = new MemoryStream();
+            IJsonWriter jsonWriter = factory.CreateJsonWriter(stream, isIeee754Compatible: false, encoding: Encoding.UTF8);
+
+            jsonWriter.StartObjectScope();
+            jsonWriter.WriteName("Foo");
+            jsonWriter.WriteValue("Bar");
+            jsonWriter.WriteName("Fizz");
+            jsonWriter.WriteValue(15.0m);
+            jsonWriter.EndObjectScope();
+
+            jsonWriter.Flush();
+            StreamReader reader = new StreamReader(stream);
+            stream.Seek(0, SeekOrigin.Begin);
+            string contents = reader.ReadToEnd();
+            Assert.Equal(@"{""Foo"":""Bar"",""Fizz"":15.0}", contents);

```

Comment:
```suggestion
            using (MemoryStream stream = new MemoryStream())
            {
                IJsonWriter jsonWriter = factory.CreateJsonWriter(stream, isIeee754Compatible: false, encoding: Encoding.UTF8);

                jsonWriter.StartObjectScope();
                jsonWriter.WriteName("Foo");
                jsonWriter.WriteValue("Bar");
                jsonWriter.WriteName("Fizz");
                jsonWriter.WriteValue(15.0m);
                jsonWriter.EndObjectScope();

                jsonWriter.Flush();
                using (StreamReader reader = new StreamReader(stream))
                {
                    stream.Seek(0, SeekOrigin.Begin);
                    string contents = reader.ReadToEnd();
                    Assert.Equal(@"{""Foo"":""Bar"",""Fizz"":15.0}", contents);
                }
            }
```

Reply:
Thanks. Will go with the succinct `using StreamReader reader - new StreamReader(stream);` version to avoid the nesting.